std/math/fp2.asm

```diff
@@ -111,7 +116,16 @@ let is_extension = |arr| match len(arr) {
 };
 
 /// Constructs an extension field element `a0 + a1 * X` from either `[a0, a1]` or `[a0]` (setting `a1`to zero in that case)
-let fp2_from_array = |arr| if is_extension(arr) { Fp2::Fp2(arr[0], arr[1]) } else { from_base(arr[0]) };
+let fp2_from_array = |arr| {
+    let _ = if !is_extension(arr) {
```

Comment:
I think you don't need `let _ = ` any more here.

Reply:
Looks like we still do! I got `Expected constraint but got ()`. Maybe `assert` should return `[]`?